keanu-project/src/main/java/io/improbable/keanu/distributions/discrete/Binomial.java

```diff
@@ -50,29 +50,22 @@ private static int sample(double p, int n, KeanuRandom random) {
     public DoubleTensor logProb(IntegerTensor k) {
         DoubleTensor logBinomialCoefficient = getLogBinomialCoefficient(k, n);
 
-        DoubleTensor logBinomial = p.pow(k.toDouble())
-            .times(
-                p.unaryMinus().plusInPlace(1.0).powInPlace(n.minus(k).toDouble())
-            ).logInPlace();
+        DoubleTensor kDouble = k.toDouble();
+        DoubleTensor nDouble = n.toDouble();
+        DoubleTensor logPPowK = kDouble.times(p.log());
+        DoubleTensor log1MinusPPowNMinusK = nDouble.minus(kDouble).timesInPlace(p.unaryMinus().plusInPlace(1.0).logInPlace());
 
-        return logBinomialCoefficient.plusInPlace(logBinomial);
+        return logBinomialCoefficient.plusInPlace(logPPowK).plusInPlace(log1MinusPPowNMinusK);
     }
 
     private static DoubleTensor getLogBinomialCoefficient(IntegerTensor k, IntegerTensor n) {
```

Comment:
This method is so much nicer, I wanna frame it and put it on the wall 😄

Reply:
I know! It felt so good to change this. I'm not sure what the problem with doing this originally was...